### ISSUE TITLE
Help: Can't include multiple directories with CLI flag, or exclude node_modules with config

### ISSUE THREAD

**Issue by vincerubinetti** [daidodo/format-imports#22]
Thanks for making this tool, it's great! This is more of a question or request for help. I would've made a discussion instead of an issue if they were enabled.

I'm basically trying to run this plugin recursively on a few select folders in my repo: `src`, `unit`, and `e2e`. Here's the setup I have:

**package.json**
```json
{
  "scripts": {
    "sort-imports": "npx format-imports --config ./import-sorter.json --log ./src",
  }
}
```

**import-sorter.json**
```json
{
  "autoFormat": "off",
  "emptyLinesBetweenGroups": 0,
  "formatExports": false,
  "wrappingStyle": "prettier",
  "groupRules": ["^vue", "^\\w", "^@\\w", "^@\\/", { "flags": "scripts" }]
}
```

I've looked through the docs and tried a few things. If I append `./unit ./e2e` to the command, like the docs say I can do, I simply get errors like `Command failed with exit code 1.`. I've also tried using the `exclude` and `excludeGlob` config options to exclude the most time consuming directories, `node_modules` and `dist`, but no matter what variation I try, the plugin runs on `node_modules` and takes an impractical amount of time.

One thing to possibly note is that I'm running this in a subdirectory in a monorepo, `/frontend`. I'm running the commands and everything relative to this directory, but in VS Code the project folder I have open is one level up, so maybe that is messing things up.

Could you advise some things to try?

**Comment by daidodo** [comment]
Thanks for the feedback!

Could you share more details about:
* The rough folder structure of your projects, including files you do or don't want format,
* The position you are running the command, and
* The effect you expect.

**Comment by vincerubinetti** [comment]
Indeed, I'll try to get back to you with a minimum reproducible example soon. I was just wondering if you could spot any errors in my config above or your code just off the cuff.

**Comment by vincerubinetti** [comment]
Here is a minimal reproducible example:

[repo.zip](https://github.com/daidodo/format-imports/files/11435954/repo.zip)

First, `cd` into `/frontend`.

I want to format `/src`, `/e2e`, `/unit`, but NOT `/node_modules`.

This works, if I just want to format `/src`:

```
npx format-imports --config ./import-sorter.json --log ./src
```

[This part of the docs](https://github.com/daidodo/format-imports#check-files-and-directories) suggests you can specify multiple directories in a single command, but apparently only with `--check` and not formatting:

```
npx format-imports --config ./import-sorter.json --log ./src ./e2e ./unit
```

It'd be nice if this worked the same way as `--check`.

As an alternative, with `"exclude": ["node_modules"]` in the config file, I should be able to do this:

```
npx format-imports --config ./import-sorter.json --log ./
```

... but this still seems to scan `node_modules` anyway. Maybe it doesn't actually format it, but it still takes a really long time and shouldn't be scanned.

**Comment by vincerubinetti** [comment]
> For 2, it's a design decision to support single directory because you can't specify `-o` if there are multiple directories. The workaround is `format-imports dir1/ && format-imports dir2/ && ...`.

This is understandable.

> > For 1, I don't think _node_modules/_ are formatted after testing your repo on my laptop. The long time is because the tool needs to find out all files (including _node_modules/_) and check if they need process.

In my repo, it takes ~2 minutes to complete scanning `node_modules` on a (very fast) M1 Macbook. I'd expect `exclude` to ignore matches completely, like a `.gitignore`, and not just ignore them for formatting.

> format-imports dir1/ && format-imports dir2/ && ...

This is fine but will get pretty lengthy and full of code duplication with a longer command: `npx format-imports --config ./import-sorter.json --log dir`.

**Comment by daidodo** [comment]
Here is the code for handling `exclude`: https://github.com/daidodo/format-imports/blob/main/src/lib/config/index.ts#L39

Happy to receive PRs if you think it can be improved.

Thanks!

**Comment by laser** [comment]
> In my repo, it takes ~2 minutes to complete scanning node_modules on a (very fast) M1 Macbook. I'd expect exclude to ignore matches completely, like a .gitignore, and not just ignore them for formatting.

I'm in the same boat. Doing something like `"exclude": ["node_modules"]` in my `import-sorter.json` file does prevent the tool from reformatting the sources in `node_modules`, but doesn't prevent the tool from scanning the directory, which takes O(minute).

**Comment by vincerubinetti** [comment]
For what it's worth, I ended up switching to a prettier plugin for this, `@ianvs/prettier-plugin-sort-imports`. It works really well for me, integrating into a tool I'm already using for formatting and testing formatting, instead of a separate command I have to run. It also makes sense to me to be a Prettier plugin since it's just formatting and doesn't affect code functionality.

I appreciate the hard work that went into this library, but that plugin ended up being a smoother experience, both with integration and configuration. I think this library has more options though. Perhaps the author here could consider porting this to a Prettier plugin.
